src/Swift.Bindings/src/Emitter/StringEmitter/Handler/TypeHandler.cs

```diff
@@ -519,16 +519,136 @@ IntPtr ISwiftObject.MarshalToSwift(IntPtr swiftDest)
+        {
+            var text = $$"""
+            private static Dictionary<Type, string> _protocolConformanceSymbols;
+
+            static {{_structDecl.Name}}()
+            {
+                _protocolConformanceSymbols = new Dictionary<Type, string>
+                {
+                    {{GenerateGetProtocolConformanceDictionaryEntries()}}
+                };
             }
             """;
 
             _writer.WriteLines(text);
             _writer.WriteLine();
+        }
+
+        private string GenerateGetProtocolConformanceDictionaryEntries()
+        {
+            var libPath = _typeDatabase.GetLibraryPath(_moduleDecl.Name);
+            var entries = new List<string>();
+            var protocolConformanceDescriptors = DemangledSymbolsRegister.Instance.GetData(libPath).ProtocolConformanceDescriptors;
 
+            foreach (var conformance in _structDecl.Conformances.Where(c => c.ProtocolSpec.Module == _moduleDecl.Name)) // Process only protocol conformances from current module for now
+            {
+                var protocol = NameProvider.GetInterfaceName(conformance.ProtocolSpec.NameWithoutModule);
+                var typeRecord = _typeDatabase.GetTypeRecordOrThrow(_moduleDecl.Name, _structDecl.FullyQualifiedNameWithoutModule);
+                var protocolConformanceSymbol = protocolConformanceDescriptors.GetValueOrDefault((new NamedTypeSpec(_structDecl.FullyQualifiedName), conformance.ProtocolSpec)); // TODO: Get rid of TypeSpec https://github.com/dotnet/runtimelab/issues/2889
+
+                entries.Add($"{{typeof({protocol}), \"{protocolConformanceSymbol}\"}}");
+            }
+
+            return string.Join(",\n", entries);
+        }
+    }
+
+    /// <summary>
+    /// Factory class for creating instances of ProtocolHandler.
+    /// </summary>
+    public class ProtocolHandlerFactory : IFactory<BaseDecl, ITypeHandler>
+    {
+        /// <summary>
+        /// Determines if the factory handles the specified declaration.
+        /// </summary>
+        /// <param name="decl">The base declaration.</param>
+        public bool Handles(BaseDecl decl)
+        {
+            return decl is ProtocolDecl;
+        }
+
+        /// <summary>
+        /// Constructs a new instance of ProtocolHandler.
+        /// </summary>
+        public ITypeHandler Construct()
+        {
+            return new ProtocolHandler();
+        }
+    }
+
+    /// <summary>
+    /// Handler class for protocol declarations.
+    /// </summary>
+    public class ProtocolHandler : BaseHandler, ITypeHandler
+    {
+        public ProtocolHandler()
+        {
+        }
+
+        /// <summary>
+        /// Marshals the specified protocol declaration.
+        /// </summary>
+        /// <param name="protocolDecl">The protocol declaration.</param>
+        /// <param name="typeDatabase">The type database instance.</param>
+        public IEnvironment Marshal(BaseDecl decl, ITypeDatabase typeDatabase)
+        {
```

Comment:
The param names in the comment don't match the signature. Is this intentional?

Reply:
Fixed